.github/workflows/validate-upstream.yml

```diff
@@ -41,24 +41,29 @@ jobs:
           path: /tmp/data/${{ inputs.data-files-folder }}
       -
         # Copy data files from /tmp/data/${{ inputs.data-files-folder }} to
-        # data/${{ inputs.data-files-folder }}. If data-files-placeholder-folder
-        # is set, then check if a placeholder file exists for each data file in
-        # that folder. If not, then creates a placeholder file with the same
-        # name as the data file, but with a .md extension.
+        # data/${{ inputs.data-files-folder }}. If create-placeholder-stubs
+        # is set to true, then check if a placeholder file exists for each data file in
+        # that folder. If not, create a placeholder stub file for the data file.
         name: Copy data files
         if: ${{ inputs.data-files-id != '' && inputs.data-files-folder != '' }}
         uses: actions/github-script@v7
         with:
           script: |
             const fs = require('fs');
             const path = require('path');
-            const dataFilesPlaceholderFolder = `content/${{ inputs.data-files-placeholder-folder }}`;
+            const glob = require('glob');
             const globber = await glob.create(`/tmp/data/${{ inputs.data-files-folder }}/*.yaml`);
             for await (const yamlSrcPath of globber.globGenerator()) {
               const yamlSrcFilename = path.basename(yamlSrcPath);
+              const yamlSrcNoExt = yamlSrcPath.replace(".yaml", "");
+              const hasSubCommands = (await (await glob.create(yamlSrcNoExt)).glob()).length > 1;
```

Comment:
This checks if there are more than one data files with the same base name, indicating that the current data file has subcommands.